### PR TITLE
dmodule: Add Package.create method

### PR DESCRIPTION
**Pull request by ibuclaw**
So it can be called from C++.

### CODE REVIEW

File: src/dmd/dmodule.d [hunk 1]
```diff
@@ -159,6 +159,11 @@ extern (C++) class Package : ScopeDsymbol
         this.tag = packageTag++;
     }
 
+    static Package create(Identifier ident)
```
Comment (jacob-carlborg) [inline src/dmd/dmodule.d]: Ideally all public symbols should have Ddoc comment.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by ibuclaw** [comment]
Actually, won't be needing this at all.  It's use I'll defer to the ldc camp.

**Comment by thewilsonator** [comment]
I don't think we need this, @kinke ?

**Comment by ibuclaw** [comment]
Or maybe I still do yet, not yet finished with what I'm attempting to do. :-)

**Comment by kinke** [comment]
LDC doesn't need this kind of code duplication in the DMD code - we use a mixin to create the D wrappers (https://github.com/ldc-developers/ldc/blob/master/dmd/ldcbindings.d) and a separate C++ header (https://github.com/ldc-developers/ldc/blob/master/dmd/ldcbindings.h).

**Comment by ibuclaw** [comment]
Yes I think this is ultimately useful without.

**Pull request closed without merging**